system/camerad/cameras/spectra.cc

```diff
@@ -1425,36 +1425,28 @@ void SpectraCamera::handle_camera_event(const cam_req_mgr_message *event_data) {
 }
 
 bool SpectraCamera::syncFirstFrame(int camera_id, uint64_t raw_id, uint64_t timestamp) {
-  std::lock_guard lk(frame_sync_mutex);
-
   if (first_frame_synced) return true;
 
   // Store the frame data for this camera
-  camera_sync_data[camera_id] = SyncData{raw_id, timestamp, raw_id + 1};
+  camera_sync_data[camera_id] = SyncData{timestamp, raw_id + 1};
 
   // Ensure all cameras are up
-  bool all_cams_up = true;
-  for (const auto& config : ALL_CAMERA_CONFIGS) {
-    if (camera_sync_data.find(config.camera_num) == camera_sync_data.end()) {
-      all_cams_up = false;
-    }
-  }
+  bool all_cams_up = camera_sync_data.size() == std::size(ALL_CAMERA_CONFIGS);
```

Comment:
how does this handle disabled cameras?

Reply:
The original code does not handle disabled cameras. Should we refactor it to support this?
```
  for (const auto& config : ALL_CAMERA_CONFIGS) {
    if (camera_sync_data.find(config.camera_num) == camera_sync_data.end()) {
      all_cams_up = false;
    }
  }
```

Reply:
yeah, need to support that